src/cursor.c

```diff
@@ -70,6 +70,36 @@ static void Cursor_FreeInternal(Cursor *cur, khiter_t khi) {
   rm_free(cur);
 }
 
+static void Cursors_ForEach(CursorList *cl, void (*callback)(CursorList *, Cursor *, void *),
+                            void *arg) {
+  for (size_t ii = 0; ii < ARRAY_GETSIZE_AS(&cl->idle, Cursor *); ++ii) {
+    Cursor *cur = *ARRAY_GETITEM_AS(&cl->idle, ii, Cursor **);
+    Cursor *oldCur = NULL;
+
+    while (cur && cur != oldCur) {
```

Comment:
why do you need this while? the cursor array might change during the run or somethign?

Reply:
Ohh I see you are removing them during the iteration.. maybe it will be better just to run from the end to the start, I think it will solve it and make the code more readable ...

Reply:
The previous approach was to run from start to end, but the issue was that half of the cursors ended up not getting deleted.

Reply:
why was it happening, scanning from start to end should not missed cursors